app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,73 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def transform
+
+      if (params[:date_coverage_1_year].length>0)
+        yyyy1= params[:date_coverage_1_year].to_i 
+      else
+        yyyy2 = 0 #error 
+      end  
+      if (!params[:date_coverage_1_month].eql?("--"))
```

Comment:
Using the hash equality `eql?` is different than the generic equality `==`.  `eql?` is generally stricter, and implemented differently in Date and Numeric classes.